libs/utils/utils.js

```diff
@@ -531,7 +547,7 @@ export function decorateImageLinks(el) {
       const pic = img.closest('picture');
       const picParent = pic.parentElement;
       if (href.includes('.mp4')) {
-        const a = createTag('a', { href: url, 'data-video-poster': img.src });
+        const a = createTag('a', { href: url, 'data-video-poster': getImgSrc(picParent) });
```

Comment:
I would shift the device detection 
```suggestion
        const a = createTag('a', { 
          href: url, 
          'data-video-poster-mobile': picParent.querySelector('source[type="image/webp"]:not([media])'),
          'data-video-poster-desktop': picParent.querySelector('source[type="image/webp"][media]')
        });
```

Then in decorate.js https://github.com/adobecom/milo/blob/stage/libs/utils/decorate.js#L179

```js
const getVideoPoster = (dataset) => {
  if(!videoPosterMobile || !videoPosterDesktop) return
  if(window.innerWidth <= 600) return `poster='${videoPosterMobile}'`
  return `poster='${videoPosterDesktop}'`
}
const poster = getVideoPoster(dataset) || ''
```

Reply:
I like this option! Going a step further, could we not just send the element that contains all the sources and let `decorate.js` decide?

On another note, I'm still not convinced about the `600px` value, even marquees start mobile styles at `768px`, could we have another look?

Reply:
I have made changes and set the video-poster dataset to the whole picture element and handled the logic in decorate.js. @overmyheadandbody : I observed that the styles for mobile start from 600px for marquee.